python/shotgun_desktop/startup.py

```diff
@@ -191,13 +192,22 @@ def __initialize_sgtk(sgtk, app_bootstrap):
         dm = sgtk.util.CoreDefaultsManager()
         sg_auth = shotgun_authentication.ShotgunAuthenticator(dm)
 
-        # get the current user
-        user = sg_auth.get_default_user()
-        logger.info("Setting current user: %r" % user)
-        sgtk.set_authenticated_user(user)
+        # get the current user for the pipeline.
+        core_user = sg_auth.get_default_user()
+
+        # If there is a user and it's a script user, use that
+        if core_user and not core_user.login:
```

Comment:
no possibility that `core_user` is a script user? I am guessing not - just double checking...